R/template.R

```diff
@@ -100,33 +100,47 @@ numeric_metric_summarizer <- function(name,
     error_call = error_call
   )
 
-  if (!quo_is_null(case_weights)) {
+  if (quo_is_null(case_weights)) {
+    case_weights <- NULL
+  } else {
     case_weights <- yardstick_eval_select(
       expr = case_weights,
       data = data,
       arg = "case_weights",
       error_call = error_call
     )
-
-    case_weights <- expr(.data[[!!case_weights]])
   }
 
-  out <- dplyr::summarise(
-    data,
-    .metric = .env[["name"]],
-    .estimator = finalize_estimator(
-      .data[[truth]],
-      metric_class = .env[["name"]]
-    ),
-    .estimate = fn(
-      truth = .data[[truth]],
-      estimate = .data[[estimate]],
-      case_weights = !!case_weights,
-      na_rm = .env[["na_rm"]],
-      !!!fn_options
+  group_rows <- dplyr::group_data(data)$.rows
+  groups_data <- vctrs::vec_chop(data, indices = group_rows)
```

Comment:
One naming scheme you could use is to call this `groups <- ` and the in the loop at each iteration you do

```r
group <- groups[[i]]
```

I like that better than the `elt` idea. I think I typically only do `elt` if the thing I'm iterating over isn't a plural word (i.e. `groups` is plural so you don't need `elt` since you can use `group`)

Then you'd have `group_truth`, `group_estimate`, and `group_case_weights`

Reply:
done in [`f272a12` (#405)](https://github.com/tidymodels/yardstick/pull/405/commits/f272a12af1d6b5ccf4b16af6598af88557737cac)